### ISSUE TITLE
[cve-2021-3121] prometheus using GoGo Protobuf before 1.3.2

### ISSUE THREAD

**Issue by simongottschlag** [penglongli/gin-metrics#15]
Hi!

We are using your great package and it makes life a lot easier! Thank you so much for maintaining it. :1st_place_medal: 

When running a security scan of our code base, we're seeing the following dependency with a known vulnerability:
```
Language: Generic
Severity: HIGH
Line: 0
Column: 0
SecurityTool: Trivy
Confidence: MEDIUM
File: /agent/_work/1/s/go.sum
Code: github.com/gogo/protobuf
Details: An issue was discovered in GoGo Protobuf before 1.3.2. plugin/unmarshal/unmarshal.go lacks certain index validation, aka the "skippy peanut butter" issue.
Installed Version: "1.1.1", Update to Version: "v1.3.2" for fix this issue.
PrimaryURL: https://avd.aquasec.com/nvd/cve-2021-3121.
Cwe Links: (https://cwe.mitre.org/data/definitions/129.html)
Type: Vulnerability
ReferenceHash: 131cf165c82f2b67996ac29097eb5c4a934c7f27bce1572426670a0aff6b8d74
```

I think you are getting it from the Prometheus packages. Would it be possible to see if you can update it?

Thank you!

**Comment by penglongli** [comment]
Hi @simongottschlag I'll update the prometheus version to v2.31.1 with new version v0.1.9 today. Thank you for pointing this out.

**Comment by penglongli** [comment]
Could `https://github.com/prometheus/client_golang/tree/v1.11.0` this version can fix the problem?

**Comment by penglongli** [comment]
Maybe just changing the version on my side can’t solve the problem. 

https://github.com/prometheus/client_golang/blob/v1.11.0/go.sum#L23

**Comment by simongottschlag** [comment]
I was sitting and digging through all dependencies last week and it seems like there's a compatibility thing going on where all packages with support for Go 1.16 is in the mod.sum - or something like that.

This may be the case we are seeing and then it most likely will be resolved when 1.18 is released and 1.16 is deprecated. 

I'm not 100% sure about this, but as far as I've been able to figure it out this is at least a partial explanation to some packages being in the sum file multiple times.

You can close this issue and I'll get back to you if it's still popping up when using 1.18.

Thank you for taking a look at it! 🎉🙏

**Comment by penglongli** [comment]
> I was sitting and digging through all dependencies last week and it seems like there's a compatibility thing going on where all packages with support for Go 1.16 is in the mod.sum - or something like that.
> 
> This may be the case we are seeing and then it most likely will be resolved when 1.18 is released and 1.16 is deprecated.
> 
> I'm not 100% sure about this, but as far as I've been able to figure it out this is at least a partial explanation to some packages being in the sum file multiple times.
> 
> You can close this issue and I'll get back to you if it's still popping up when using 1.18.
> 
> Thank you for taking a look at it! 🎉🙏

You are welcome, I will continue to pay attention to this problem. Thank you for pointing this out. 🎉
